src/compile/data/index.ts

```diff
@@ -43,7 +44,23 @@ export interface DataComponent {
   isFaceted: boolean;
 
   /**
-   * Parse properties passed down from ancestors.
+   * Parse properties passed down from ancestors. Helps us to keep track of what has been parsed or is derived.
    */
-  ancestorParse: Dict<string>;
+  ancestorParse?: AncestorParse;
+}
+
+export class AncestorParse extends Split<Dict<string>> {
```

Comment:
Why call it AncestorParse?  I don't see anything that is specific about Ancestor. 

ParseComponent is probably more appropriate and consistent with the rest of the code base.

Reply:
This already existed: https://github.com/vega/vega-lite/blob/master/src/compile/data/index.ts#L48. Let's rename things in a new PR.

Reply:
Your link only shows an old `Dict<string>`, but this class is introduced in this PR so naming it property should be in the scope of this PR.

Reply:
ParseComponent is not more appropriate, though because we never generate a parse object from this. I added a comment to make this clear.